### PR TITLE
Updates to Gradle

### PR DESCRIPTION
**Pull request by citizenmatt**
Some updates to the Gradle files:

- [x] Add the `kotlin` source folder to the `kotlin.srcDir` property, rather than the Java one. Also means that the `java` folder isn't removed from the `java.srcDir` property, and the plugin can compile both Kotlin and Java
- [x] Update Gradle/Gradle wrapper to latest (6.8.3)
- [x] Update dependencies to latest versions
- [x] Update plugin management
- [x] Provide some defaults for `runIde` task
- [x] Prevent Kotlin 1.4 bundling the stdlib in the built plugin
- [x] Fix some deprecated Gradle warnings

### CODE REVIEW

**Comment by citizenmatt** [comment]
Looking at the gradle scripts, should the `compileDotNet` task be calling `msbuild /t:Restore;Rebuild`? Should it just call `msbuild /t:Restore;Build`? A rebuild will clean before building, which is probably unnecessary.

**Review by ForNeVeR — APPROVED** [review]

**Comment by matkoch** [comment]
Merged and published with version `2021.1.0` of the template. Thanks <3

**Comment by matkoch** [comment]
Regarding the `Rebuild` I will check again. I barely remember, but I think it actually was for a reason (which could even be outdated). I will try during my active plugin development.

File: content/build.gradle [hunk 1]
```diff
@@ -122,12 +114,17 @@ intellij {
 }
 
 runIde {
-    // TODO: adapt runtime if necessary
+    // Match Rider's default heap size of 1.5Gb (default for runIde is 512Mb)
+    maxHeapSize = "1500m"
```
Comment (auduchinok) [inline content/build.gradle:118]: Could this be moved to `gradle-intellij`?

File: content/build.gradle [hunk 2]
```diff
@@ -122,12 +114,17 @@ intellij {
...
+
+    // Rider's backend doesn't support dynamic plugins. It might be possible to work with auto-reload of the frontend
+    // part of a plugin, but there are dangers about keeping plugins in sync
+    autoReloadPlugins = false
```
Comment (auduchinok) [inline content/build.gradle:122]: Probably the same as above.

**Pull request merged**